src/App.tsx

```diff
@@ -1,24 +1,205 @@
+      loadingTodos();
+    } catch (error) {
+      showError(ErrorNotice.DELETE);
+    } finally {
+      setProcessingIDs([0]);
+    }
+  }, [todosFromServer]);
+
+  const handleDeleteCompleted = useCallback(
+    async (todos: Todo[]) => {
+      try {
+        setProcessingIDs(todos.map(todo => todo.id));
+        const removeAllComleted = todos.map(todo => (
+          deleteTodo(todo.id)
+        ));
+
+        await Promise.all(removeAllComleted);
+
+        loadingTodos();
+      } catch (error) {
+        showError(ErrorNotice.DELETE);
+      } finally {
+        setProcessingIDs([0]);
+      }
+    }, [inProcessing],
+  );
+
+  const handleComplete = useCallback(
+    async (todoId: number[], data: boolean) => {
+      setProcessingIDs(todoId);
+
+      try {
+        const completingTodos = todoId.map(id => (
+          completeTodo(id, data)
+        ));
+
+        await Promise.all(completingTodos);
+
+        loadingTodos();
+      } catch (error) {
+        showError(ErrorNotice.UPDATE);
+      } finally {
+        setProcessingIDs([0]);
+      }
+    }, [todosFromServer],
+  );
+
+  const changedTitle = useCallback(async (id: number, newTitle: string) => {
+    setProcessingIDs([id]);
+
+    try {
+      await renameTodo(id, newTitle);
+      loadingTodos();
+    } catch (error) {
+      setMessageError(ErrorNotice.UPDATE);
+    } finally {
+      setProcessingIDs([0]);
+    }
+  }, [todosFromServer]);
+
+  const visibleTodos = useMemo(() => (
+    getVisibleTodos(todosFromServer, filter)
+  ), [todosFromServer, filter]);
+
+  const completedTodos = useMemo(
+    () => todosFromServer.filter(todo => (todo.completed)),
+    [todosFromServer, tempTodo],
+  );
+
+  const activeTodos = useMemo(
+    () => todosFromServer.filter(todo => !todo.completed),
+    [todosFromServer],
+  );
+
   if (!USER_ID) {
     return <UserWarning />;
   }
 
   return (
-    <section className="section container">
-      <p className="title is-4">
-        Copy all you need from the prev task:
-        <br />
-        <a href="https://github.com/mate-academy/react_todo-app-add-and-delete#react-todo-app-add-and-delete">React Todo App - Add and Delete</a>
-      </p>
-
-      <p className="subtitle">Styles are already copied</p>
-    </section>
+    <div className="todoapp">
+      <h1 className="todoapp__title">todos</h1>
+
+      <div className="todoapp__content">
+        <Header
+          isDisableInput={isDisableInput}
+          title={title}
+          handleSubmit={handleSubmit}
+          createNewTitle={setNewTitle}
+          completedTodos={completedTodos}
+          activeTodos={activeTodos}
+          updateTodos={handleComplete}
+        />
+
+        <TodoList
+          todos={visibleTodos}
+          creating={tempTodo}
+          onDelete={handleDelete}
+          inProcessing={inProcessing}
+          updateTodo={handleComplete}
+          onChangeTitle={changedTitle}
+        />
+
+        {!!todosFromServer.length && (
```

Comment:
```suggestion
        {todosFromServer.length > 0 && (
```
https://mate-academy.github.io/style-guides/javascript.html#153use-shortcuts-for-booleans-but-explicit-comparisons-for-strings-and-numbers